netty/src/main/java/io/grpc/netty/NettyHandlerSettings.java

```diff
@@ -56,17 +63,23 @@ public static synchronized void autoWindowOn(boolean autoFlowControl) {
     autoFlowControlOn = autoFlowControl;
   }
 
-  public static synchronized int getLatestClientWindow() {
-    return getLatestWindow(clientHandler);
+  public static synchronized Future<Integer> getLatestClientWindow() {
+    return getLatestWindow(clientLoop, clientHandler);
```

Comment:
`checkNotNull(clientLoop, "setAutoWindow must be called first")` ? 

Also I don't see the point to synchronized it, unless `setAutoWindow` can be reset for multiple times in different threads which does not make sense and is make trouble.

Reply:
since this method doesn't actually use the method, I am reluctant to put the check here.  I put it down where it is used.